### PR TITLE
Refactor / replace os.path with pathlib in many places

### PR DESCRIPTION
**Pull request by jlnav**
Should make workflow/ensemble/sim dirs easier to read and maintain.

### CODE REVIEW

File: libensemble/tests/scaling_tests/forces/forces_gpu/run_libe_forces.py [hunk 1]
```diff
@@ -24,11 +24,12 @@
 
 # Fixed resources (one resource set per worker)
 from libensemble.gen_funcs.sampling import uniform_random_sample as gen_f
+from libensemble.libE import libE
```
Comment (shuds13) [inline libensemble/tests/scaling_tests/forces/forces_gpu/run_libe_forces.py:27]: I don't really get this change. The gen line for var resources was next to the other one.

Comment (jmlarson1) [reply libensemble/tests/scaling_tests/forces/forces_gpu/run_libe_forces.py:27]: This is what `isort` prefers.

File: libensemble/tests/scaling_tests/forces/forces_gpu/run_libe_forces.py [hunk 2]
```diff
@@ -24,11 +24,12 @@
...
+from libensemble.tools import add_unique_random_streams, parse_args
```
Comment (shuds13) [inline libensemble/tests/scaling_tests/forces/forces_gpu/run_libe_forces.py:28]: i prefer libE as first import.

File: libensemble/tests/unit_tests/test_loc_stack.py [hunk 3]
```diff
@@ -56,7 +57,8 @@ def test_location_stack():
 
         # Push registered location (we should move
         s.push_loc(0)
-        assert s.stack == [None, start_dir], "Directory stack is incorrect." "Wanted [None, {}], got {}.".format(
+        stack = s.stack
+        assert stack == [None, Path(start_dir)], "Directory stack is incorrect." "Wanted [None, {}], got {}.".format(
```
Comment (shuds13) [inline libensemble/tests/unit_tests/test_loc_stack.py:61]: Isn't Path already applied? Is this passing?

Comment (shuds13) [reply libensemble/tests/unit_tests/test_loc_stack.py:61]: Ok, it passes, but I dont think its needed

**Comment by shuds13** [comment]
Will need to resolve comments/conflicts then I approve this.

**Pull request merged**